react: bump version to latest

vcapretz opened the pull request:
this could be a simple first step, between 15 and 16 there are no major breaking changes
for that basically the only thing needed was to add some polyfills

the UNSAFE methods that are renamed should still be working as usual, but react will remove then on 17
so it's a good idea to refactor the components that now have the UNSAFE methods
to read more about it: https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

also, with this new version we can start using React Hooks to write functional components:
https://reactjs.org/docs/hooks-state.html#hooks-and-function-components

some other style changes were included since my Prettier setup tried figuring out stuff by itself,
it's probably a good idea to officially add Prettier to this project so this wouldn't happen in the future

# This PR is connected to a issue?

connected to #335 

## This PR needs some special attention?

I reviewed the changelog for react https://github.com/facebook/react/blob/master/CHANGELOG.md 
and things should still be working fine, I tested out locally and didn't face any problems 
but ofc we don't live in a perfect world so some bugs _might_ be introduced, so it's a good idea to have 
more people's eyes in here and find them 😄

vcapretz committed:
react: bump version to latest

this could be a simple first step, between 15 and 16 there are no major breaking changes
for that basically the only thing needed was to add some polyfills

the UNSAFE methods that are renamed should still be working as usual, but react will remove then on 17
so it's a good idea to refactor the components that now have the UNSAFE methods
to read more about it: https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

also, with this new version we can start using React Hooks to write functional components:
https://reactjs.org/docs/hooks-state.html#hooks-and-function-components

some other style changes were included since my Prettier setup tried figuring out stuff by itself,
it's probably a good idea to officially add Prettier to this project so this wouldn't happen in the future

vcapretz commented on frontend/src/containers/Home/Home.js:
```diff
@@ -120,20 +127,20 @@ class Home extends React.Component {
-        }
-        {this.state.request &&
+        )}
+        {this.state.request && (
```
will revert the changes that prettier formated since the codacy won't like them 😅

lucassabreu replied:
we are using the rules in the `frontend/.eslintrc` to format and validate js, prettier kinda of uses its own rules, so would be better for you to use eslint instead, and probability things will go smoothly

vcapretz replied:
yep, I will just disable prettier locally 👍 will send another commit soon

vcapretz commented on frontend/src/components/UploadImages/UploadImages.js:
```diff
@@ -96,11 +94,13 @@ class UploadImages extends React.Component {
-            file,
-          })}
+          customRequest={({ onSuccess, onError, file }) =>
+            this.uploadImage({
```
will revert the changes that prettier formated since the codacy won't like them 😅

vcapretz commented on frontend/src/containers/Results/Results.js:
```diff
@@ -11,9 +11,9 @@ class Results extends React.Component {
+  };
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
```
not sure on how to solve these ones for placing order, since the bot now won't recognize they are lifecycle methods and the ordering shouldn't change.
Maybe it's worth actually removing these methods and refactor a bit the components according to what React recommends now. 

wdyt?

vcapretz commented:
@lucassabreu ok it's green now 🎉 

I've changed from the `UNSAFE` methods to either putting logic inside `componentDidMount` or `constructor`, depending on each case.

Also reverted most of the prettier changes, still left some small ones like adding semicolons and removing extra spaces! 

Let me know if there's something else I can do! 😄

lucassabreu commented:
> @lucassabreu ok it's green now 
> 
> I've changed from the `UNSAFE` methods to either putting logic inside `componentDidMount` or `constructor`, depending on each case.
> 
> Also reverted most of the prettier changes, still left some small ones like adding semicolons and removing extra spaces!
> 
> Let me know if there's something else I can do!

:top: now I hand over to @RaonyMarcondes hehe
